tests/unit/commands/sync/test_sync_context.py

```diff
@@ -7,31 +7,99 @@
 
 from samcli.commands.sync.sync_context import (
     SyncState,
+    ResourceSyncState,
+    datetime,
     _sync_state_to_toml_document,
+    HASH,
+    SYNC_TIME,
     SYNC_STATE,
     DEPENDENCY_LAYER,
+    RESOURCE_SYNC_STATES,
     _toml_document_to_sync_state,
     SyncContext,
 )
 from samcli.lib.build.build_graph import DEFAULT_DEPENDENCIES_DIR
 
+MOCK_TIME = datetime(2023, 2, 8, 12, 12, 12)
+
 
 class TestSyncState(TestCase):
-    @parameterized.expand([(True,), (False,)])
-    def test_sync_state(self, dependency_layer):
-        sync_state = SyncState(dependency_layer)
+    @parameterized.expand(
+        [
+            (True, {"MockResourceId": ResourceSyncState("mock-hash", MOCK_TIME)}),
+            (False, {"MockResourceId": ResourceSyncState("mock-hash", MOCK_TIME)}),
+            (True, {"Parent/Child/MockResourceId": ResourceSyncState("nested-mock-hash", MOCK_TIME)}),
+            (
+                False,
+                {
+                    "MockResourceId": ResourceSyncState("mock-hash", MOCK_TIME),
+                    "Parent/Child/MockResourceId": ResourceSyncState("nested-mock-hash", MOCK_TIME),
+                },
+            ),
+        ]
+    )
+    def test_sync_state(self, resource_sync_states, dependency_layer):
```

Comment:
Just double checking, are the function arguments reversed here?

Reply:
Good catch! weird that the tests didn't fail when the wrong type args were provided.